### PR TITLE
Automatically detect generated code configuration

### PR DESCRIPTION
**Pull request by sharwell**
This change eliminates the need to specify `SkipGeneratedCodeCheck` when the analyzer calls `ConfigureGeneratedCodeAnalysis` explicitly.

### CODE REVIEW

File: THIRD-PARTY-NOTICES.txt [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+distributed under licenses different than the Roslyn SDK software.
+
+In the event that we accidentally failed to list a required notice, please
+bring it to our attention. Post an issue or email us:
+
+           dotnet@microsoft.com
+
+The attached notices are provided for information only.
+
+License notice for StreamJsonRpc
```
Comment (jmarolf) [inline THIRD-PARTY-NOTICES.txt:11]: Is this change unrelated to the PR? If so please do this in a separate PR. I do not see streamjson RPC being used anywhere

Comment (sharwell) [reply THIRD-PARTY-NOTICES.txt:11]: A bunch of the dynamic code in `AnalyzerInfo` was derived from the proxy in StreamJsonRpc. I'm checking with @AArnott to see if we need to include this notice or if it was entirely authored here.

Comment (jmarolf) [reply THIRD-PARTY-NOTICES.txt:11]: ah, I see it was unclear to me that was the reason

Comment (AArnott) [reply THIRD-PARTY-NOTICES.txt:11]: The only code I recognize is the `SkipVisibilityChecksFor` and related code. But that code didn't originally come from streamjsonrpc -- it came from vs-mef:

https://github.com/microsoft/vs-streamjsonrpc/blob/d64c14cc238cf05f99c981a0ee3d0de2b168fe1f/src/StreamJsonRpc/SkipClrVisibilityChecks.cs#L4

It was entirely authored within Microsoft. So IMO we don't need to list it under 3rd party notices.
